main/src/addins/CSharpBinding/MonoDevelop.CSharp/CSharpPathedDocumentExtension.cs

```diff
@@ -614,5 +618,32 @@ void CancelUpdatePath ()
 			src.Cancel ();
 			src = new CancellationTokenSource ();
 		}
+
+		class OwnerProjectInfo
```

Comment:
```suggestion
		sealed class OwnerProjectInfo
```

Reply:
Maybe, since we implemented Equals and GetHashCode, we should also:
```suggestion
		class OwnerProjectInfo : IEquatable<T>
```